### ISSUE TITLE
Support for pre-configured external coverage packages.

### ISSUE THREAD

**Issue by wellwelwel** [wellwelwel/poku#802]
Support for the `--coverage` flag, that will require one of the following coverage dependencies:

- [**c8**](https://github.com/bcoe/c8?tab=readme-ov-file)
- [**nyc**](https://github.com/istanbuljs/nyc)
- [**monocart-coverage-reports**](https://github.com/cenfun/monocart-coverage-reports)

---

If the user doesn't use a configuration, **Poku** should use a default configuration for each one.

**Comment by DevJoaoLopes** [comment]
@wellwelwel The intention here is to use only one of the 3 libs options for the project (since even c8 is already installed as a dependency) or leave optional so that the user can choose between any of these 3 ?

**Comment by wellwelwel** [comment]
> The intention here is to use only one of the 3 libs options for the project (since even c8 is already installed as a dependency) or leave optional so that the user can choose between any of these 3 ?

@DevJoaoLopes, in both `v2` and `v3`, you can choose any package you like and that will never change. This feature's idea is to create pre-configurations for the packages mentioned, but you could choose either one of them or a completely different one 🙋🏻‍♂️

**Comment by DevJoaoLopes** [comment]
Understood, very cool :rocket: 
I want to support this development, thank you for your attention @wellwelwel

**Comment by DevJoaoLopes** [comment]
@wellwelwel Sorry for the delay in getting back to the topic, you know how hard it is to find time 😢 

I started to outline the development (there's still a lot to do), and I left everything documented in the #996 .

**Comment by mrazauskas** [comment]
Just to add my two cents, it would be really beneficial to have an easy way to collect test coverage across Node.js, Bun and Deno. (Would be beneficial for `poku` as well, or?)

---

Obviously, the native V8 coverage tools will not work on Bun. (Not sure if Deno supports these either.) But that does not mean they should be excluded or so. I think, V8 coverage combined with `monocart` tooling is the best thing available at the moment.

By the way, there is one more new V8 coverage tool: [One Double Zero](https://www.npmjs.com/package/one-double-zero).

**Comment by wellwelwel** [comment]
> Would be beneficial for `poku` as well, or?

Yes, definitely! **Poku** itself lacks coverage in these runtimes.

By suggesting an exclusive package for each coverage tool in #996 (e.g., `@pokujs/c8`), the idea is to keep scope for a possible package of its own, possibly working independently of the `poku` test runner.

---

> By the way, there is one more new V8 coverage tool: [One Double Zero](https://www.npmjs.com/package/one-double-zero).

Thanks for the tip. I'll try it out 🙋🏻‍♂️
